CMakeLists.txt

```diff
@@ -2,51 +2,87 @@ project(libMCNP2CAD)
+
+# Adjust compiler setting for Linux using gcc version 5.0 and higher
+if (CMAKE_SYSTEM_NAME MATCHES Linux AND CMAKE_COMPILER_IS_GNUCXX)
+  if (CMAKE_CXX_COMPILER_VERSION VERSION_GREATER "5.0")
+    if (NOT CMAKE_CXX_FLAGS MATCHES _GLIBCXX_USE_CXX11_ABI)
       set(CMAKE_CXX_FLAGS "-D_GLIBCXX_USE_CXX11_ABI=0 ${CMAKE_CXX_FLAGS}")
-    endif()
-  endif()
-endif()
-
-if(${BUILD_CLI})
-    add_subdirectory(${CMAKE_SOURCE_DIR}/cli)
-endif()
-
-find_package(Eigen3 3.3 REQUIRED NO_MODULE)
-
-# These directories need to be defined by the user.
-include_directories(${IGEOM_INCLUDE_DIR})
-find_library(IGEOM_LIB NAMES iGeom HINTS ${IGEOM_LIB_DIR})
-
-SET(LIB_SRC
-    mcnp2cad.cpp
-    mcnp2cad.hpp
-    MCNPInput.cpp
-    MCNPInput.hpp
-    geometry.cpp
-    geometry.hpp
-    volumes.cpp
-    volumes.hpp
-    GQ_Characterize.cpp
-    GQ_Characterize.hpp
-    test_GQ.cpp
-)
-
-add_library(mcnp2cad SHARED ${LIB_SRC})
-target_link_libraries(mcnp2cad ${IGEOM_LIB})
-
-add_executable(test_GQ test_GQ.cpp GQ_Characterize.cpp)
-target_link_libraries(test_GQ)
-if(IGEOM_LIB_DIR)
-    set_target_properties(test_GQ PROPERTIES INSTALL_RPATH ${IGEOM_LIB_DIR} BUILD_WITH_INSTALL_RPATH TRUE)
-elseif(CMAKE_PREFIX_PATH)
-    set_target_properties(test_GQ PROPERTIES INSTALL_RPATH ${CMAKE_PREFIX_PATH} BUILD_WITH_INSTALL_RPATH TRUE)
-endif()
-
-enable_testing()
-add_test(NAME test_gq_characterization COMMAND test_GQ)
+    endif ()
+  endif ()
+endif ()
+
+if (BUILD_CLI)
+  add_subdirectory(cli)
+endif ()
+
+# IGEOM_DIR must be specified if mcnp2cad is being built as a standalone
+# library. If it is not specified, then CMake will assume that mcnp2cad is being
+# built as a part of a larger project which already knows the location of iGeom.
+if (IGEOM_DIR)
+  message(STATUS "Building standalone mcnp2cad")
+  set(STANDALONE ON)
+  include_directories(${IGEOM_DIR}/include)
+  find_library(IGEOM_LIB NAMES iGeom HINTS ${IGEOM_DIR}/lib)
+  get_filename_component(IGEOM_LIB_DIR ${IGEOM_LIB} DIRECTORY)
+  link_directories(${IGEOM_LIB_DIR})
+  message(STATUS "IGEOM_LIB_DIR: ${IGEOM_LIB_DIR}")
+else ()
+  set(STANDALONE OFF)
+endif ()
+
+# RPATH handling. If CMAKE_INSTALL_RPATH_USE_LINK_PATH is set to ON, then all
+# directories that have been linked via the link_directories() command will be
+# added to the RPATH.
+if (STANDALONE)
+  set(CMAKE_INSTALL_RPATH_USE_LINK_PATH ON)
+endif ()
+
+SET(MCNP2CAD_SRC_FILES geometry.cpp
+                       GQ_Characterize.cpp
+                       mcnp2cad.cpp
+                       MCNPInput.cpp
+                       volumes.cpp)
+
+set(MCNP2CAD_PUB_HEADERS dataref.hpp
+                         geometry.hpp
+                         GQ_Characterize.hpp
+                         mcnp2cad.hpp
+                         MCNPInput.hpp
+                         options.hpp
+                         version.hpp
+                         volumes.hpp)
+
+# mcnp2cad library
+add_library(mcnp2cad SHARED ${MCNP2CAD_SRC_FILES})
```

Comment:
How does this build find the `iGeom` libraries when part of the plugin?

Reply:
Since this CMakeFile will be included via the `add_subdirectory` command, CMake should already know what is meant when the `iGeom` library is specified. It's no different than in DAGMC, when we build the `dagmc` library, and everywhere else in the project we only need to specify `dagmc` to link to that library.

Reply:
I know this was one of the complications of the circular system... perhaps it's only the `iGeom` headers that need to be found when building as part of the plugin.  This is were there is a gap between 
* the `add_subdirectory` implementation in which it is difficult to pass in the location of headerse
* the build directly from the parent directory in which you can take full control but may diverge from this package's build configuration

It would be great if there was a clean CMake-approved way to pass information into this build using the `add_subdirectory` method

Reply:
I'm not sure if I'm following what the problem is

Reply:
I'll have to try it to articulate more clearly, but using the `add_subdirectory` method from the plugin will drop into this directory and run it's CMake.  Building this library requires access to `iGeom` headers, but those headers are not installed anywhere and I am not sure if there is a way to alter the list of include directories for this project from the CMake instance that calls it.

Reply:
When we build `mcnp2cad` as a part of the plugin, it knows where the headers are because we're also building `iGeom` as a part of the plugin. The top-level Trelis-plugin CMakeFile includes the line

```
include_directories(iGeom)
```

which means that all CMakeFiles in subdirectories already have the iGeom headers available.

Reply:
OK - I'll just have to try it.  I know there was something not working correctly at some point...